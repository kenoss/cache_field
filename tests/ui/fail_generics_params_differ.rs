#[struct_cache_field::impl_cached_method]
impl<T> Hoge<T>
where
    T: ToString + From<String>,
{
    pub fn two_times_x(&self) -> T {
        let s = self.x.to_string();
        format!("{s}{s}").into()
    }
}

#[struct_cache_field::add_cache_field]
struct Hoge<S>
where
    S: ToString + From<String>,
{
    x: u64,
    t: S,
}

fn main() {}
