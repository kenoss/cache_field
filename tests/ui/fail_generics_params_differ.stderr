error: generics differ, which must coincide as string:
           in impl cached methods: < T > where T : ToString + From < String > ,
           in struct definition:   < S > where S : ToString + From < String > ,
  --> tests/ui/fail_generics_params_differ.rs:13:12
   |
13 | struct Hoge<S>
   |            ^^^

error[E0609]: no field `__cache_fields__` on type `&Hoge<T>`
 --> tests/ui/fail_generics_params_differ.rs:1:1
  |
1 | #[struct_cache_field::impl_cached_method]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ unknown field
  |
  = note: this error originates in the attribute macro `struct_cache_field::impl_cached_method` (in Nightly builds, run with -Z macro-backtrace for more info)
